src/AppBundle/Controller/BlogController.php

```diff
@@ -114,4 +116,43 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ));
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $q = $request->query->get('q');
+
+        // Sanitizing the query: removes all non-alphabetic symbols
+        $query = preg_replace('/[^[:alnum:] ]/', '', trim(preg_replace('/[[:space:]]+/', ' ', $q)));
+
+        // Splits the query into terms and removes all terms which
+        // length is less than 2
+        $terms = array_unique(explode(' ', strtolower($query)));
+        $terms = array_filter($terms, function($term) {
+            return 2 <= strlen($term);
+        });
+
+        $posts = new ArrayCollection();
+
+        if (!empty($terms)) {
+            $em = $this->getDoctrine()->getManager();
```

Comment:
@theofidry Why do you think so? We get exactly the **entity** manager here, so I prefer to use `$em`, it's more specific I think.